src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultStates = [];
+
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let newState = { ...currentState };
```

Comment:
[NAMING] - use proper names for object copy. Instead of `newState`, a more descriptive name like `stateCopy` would convey the purpose of the variable more clearly.